flow/flow.go

```diff
@@ -989,6 +989,9 @@ func (f *Flow) Canonicalize(config Config) *Flow {
 
 func (f *Flow) canonicalize(m *flowMap, config Config) *Flow {
 	if flow := m.Get(f); flow != nil {
+		if f.MustIntern && !flow.MustIntern {
+			return f
```

Comment:
i think this should not return but rather fall through, right? we want to register it as canonical [1], and we do want to canonicalize multiple interns that do have MustIntern set.

this leaves a bit of a conundrum: the hash function in flowMap must account for MustIntern, though that's a simple modification in *flowMap put/get.

[1] and also canonicalize it with respect to its children, though this always being an intern it's not an issue; though this should at least be documented.

Reply:
I have modified the flow digest to include forced interns instead of just the flowmap digest computation. Now that I think about it, I am not sure modifying the flow digest is such a great idea. Just changing the flow map hash seems sufficient and correct. I'll push another diff shortly. let me know, in the meantime, what you think.

Reply:
done